[helm v3] The lock file (requirements.lock) is out of sync when using repository alias

This is similar to #7233.
`helm dependency build` using Helm 3 with v1 API fails with the following when repository in `requirements.yaml` is referenced by an **alias**.
```bash
manager.go:94: warning: a valid Helm v3 hash was not found. Checking against Helm v2 hash...
Error: the lock file (requirements.lock) is out of sync with the dependencies file (requirements.yaml). Please update the dependencies
```

### helm version
```bash
$ helm version --client
Client: &version.Version{SemVer:"v2.14.3", GitCommit:"0e7f3b6637f7af8fcfddb3d2941fcc7cbebb0085", GitTreeState:"clean"}
$ helm3 version
version.BuildInfo{Version:"v3.1+unreleased", GitCommit:"", GitTreeState:"clean", GoVersion:"go1.13.8"}
```

### steps to reproduce
#### helm v2
```bash
$ helm create test-chart
Creating test-chart
$ cd test-chart/
$ helm repo add elastic https://helm.elastic.co
"elastic" has been added to your repositories
$ cat << EOF > requirements.yaml
dependencies:
  - name: elasticsearch
    version: 7.5.1
    repository: '@elastic'
EOF
$ helm dep up
Hang tight while we grab the latest from your chart repositories...
...Unable to get an update from the "local" chart repository (http://127.0.0.1:8879/charts):
	Get http://127.0.0.1:8879/charts/index.yaml: dial tcp 127.0.0.1:8879: connect: connection refused
...Successfully got an update from the "elastic" chart repository
...Successfully got an update from the "stable" chart repository
Update Complete.
Saving 1 charts
Downloading elasticsearch from repo https://helm.elastic.co
Deleting outdated charts
```

#### helm v3
```bash
$ helm3 repo add elastic https://helm.elastic.co
"elastic" has been added to your repositories
$ helm3 dep build
manager.go:94: warning: a valid Helm v3 hash was not found. Checking against Helm v2 hash...
Error: the lock file (requirements.lock) is out of sync with the dependencies file (requirements.yaml). Please update the dependencies
```
**Note** the same code works fine when repository in `requirements.yaml` is referenced by a full URL (e.g. https://helm.elastic.co)

--- a/pkg/downloader/manager.go
+++ b/pkg/downloader/manager.go
@@ -82,6 +82,19 @@ func (m *Manager) Build() error {
 
 	// Check that all of the repos we're dependent on actually exist.
 	req := c.Metadata.Dependencies
+
+	// If using apiVersion v1, calculate the hash before resolve repo names
+	// because resolveRepoNames will change req if req uses repo alias
+	// and Helm 2 calculate the digest from the original req
+	// Fix for: https://github.com/helm/helm/issues/7619
+	var v2Sum string
+	if c.Metadata.APIVersion == chart.APIVersionV1 {
+		v2Sum, err = resolver.HashV2Req(req)
+		if err != nil {
+			return errors.New("the lock file (requirements.lock) is out of sync with the dependencies file (requirements.yaml). Please update the dependencies")
+		}
+	}
+
 	if _, err := m.resolveRepoNames(req); err != nil {
 		return err
 	}
@@ -92,7 +105,7 @@ func (m *Manager) Build() error {
 		// Fix for: https://github.com/helm/helm/issues/7233
 		if c.Metadata.APIVersion == chart.APIVersionV1 {
 			log.Println("warning: a valid Helm v3 hash was not found. Checking against Helm v2 hash...")
-			if sum, err := resolver.HashV2Req(req); err != nil || sum != lock.Digest {
+			if v2Sum != lock.Digest {
 				return errors.New("the lock file (requirements.lock) is out of sync with the dependencies file (requirements.yaml). Please update the dependencies")
 			}
 		} else {

